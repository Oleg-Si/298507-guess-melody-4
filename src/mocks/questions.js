const AVATAR_URL = `//picsum.photos/134/134?r=`;

export default [
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        id: 1,
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
        genre: `rock`,
      },
      {
        id: 2,
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
        genre: `blues`,
      },
      {
        id: 3,
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
        genre: `jazz`,
      },
      {
        id: 4,
        src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
        genre: `rock`,
      }
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    },
    answers: [
      {
        id: 1,
        picture: `${AVATAR_URL}${Math.random()}`,
        artist: `Chivas Regal`,
      },
      {
        id: 2,
        picture: `${AVATAR_URL}${Math.random()}`,
        artist: `Jack Daniels`,
      },
      {
        id: 3,
        picture: `${AVATAR_URL}${Math.random()}`,
        artist: `Jim Beam`,
      }
    ],
  }
];
