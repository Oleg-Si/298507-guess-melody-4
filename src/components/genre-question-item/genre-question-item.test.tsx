import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {questionGenreForTest} from '../../mocks/questions';
import GenreQuestionItem from './genre-question-item';

it(`GenreQuestionItem is rendered correctly`, () => {
  const tree = renderer.create(
      <GenreQuestionItem
        id={0}
        answer={questionGenreForTest.answers[0]}
        onChange={() => null}
        renderPlayer={() => null}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
