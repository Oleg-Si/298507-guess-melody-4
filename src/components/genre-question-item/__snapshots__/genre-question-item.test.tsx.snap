// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionItem is rendered correctly 1`] = `
<div
  className="track"
>
  <div
    className="game__answer"
  >
    <input
      className="game__input visually-hidden"
      id="answer-0"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="rock"
    />
    <label
      className="game__check"
      htmlFor="answer-0"
    >
      Отметить
    </label>
  </div>
</div>
`;
